src/Model/ModelManagerInterface.php

```diff
@@ -13,6 +13,7 @@
 
 namespace Sonata\AdminBundle\Model;
 
+use Doctrine\Common\Collections\Collection;
```

Comment:
This is more coupling to the Doctrine thing, but what should we do? On SonataDoctrineOrmAdminBundle the modelManager expects a Collection, because it is using methods from it. Thoughts?

We can leave it as before, but when fixing the typings on SonataDoctrineOrmAdmin we will have to type that as Collection

Reply:
I have created an issue #6024 about this.

If we add the Collection typehint, these methods seems fully useless because they already exists in the Collection interface.

So to me we should chose lne of three solutions
- removing the collection methods in the ModelManager and assume the fact we are using Collection.
- keeping array as typehint and modifying the implementation in the storage bundle
- creating our own Collection interface we the methods add, remove, etc and removing the method from the modelmanager


The third is maybe not a bad idea wdyt ?

Reply:
I am not sure about the solution, IMO we should avoid calling toArray on doctrine collection. does it kill the performance optimization on those lazy collections? I do not know enough of doctrine 😅

Reply:
IMO, there is no issue in using classes from the `Doctrine\` NS, we should not see that as a bad thing. There are features and utilities that are agnostic to the persistence model or the model management at all.

Reply:
>I am not sure about the solution, IMO we should avoid calling toArray on doctrine collection. does it kill the performance optimization on those lazy collections? I do not know enough of doctrine sweat_smile

Regarding this concern, I think this is something about PHP instead of Doctrine. `Collection` implements `IteratorAggregate`, which allows the items in the collection to be traversed one by one if they are provided by a [generator](https://www.php.net/manual/en/class.generator.php), without loading the entire collection in memory like occurs with an array.

Reply:
> IMO, there is no issue in using classes from the `Doctrine\` NS, we should not see that as a bad thing. There are features and utilities that are agnostic to the persistence model or the model management at all.

Then we should deprecate/remove the createCollection/collectionAdd/collectionRemove/... methods of the ModelManager.

Reply:
I added the 4.0 milestone to this https://github.com/sonata-project/SonataAdminBundle/issues/6024